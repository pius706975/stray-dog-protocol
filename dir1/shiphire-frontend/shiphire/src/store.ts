import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
