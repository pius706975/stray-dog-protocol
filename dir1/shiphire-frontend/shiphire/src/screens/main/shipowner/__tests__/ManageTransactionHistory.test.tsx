import { NavigationContainer } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import MockAdapter from 'axios-mock-adapter';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import httpRequest from '../../../../services/api';
import store from '../../../../store';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { MainOwnerStackParamList } from '../../../../types';
import {
    EditShipHistory,
    ManageTransactionHistory,
} from '../manageTransactionHistory';
import { setMockDocumentPickerResolve } from '../../../../jest/setup';
import { act } from 'react-test-renderer';

const queryClient = new QueryClient();
const MainOwnerStack = createNativeStackNavigator<MainOwnerStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const mockParams = {
    shipId: '65a8bda44b191011546c21b1',
    shipHistory: [
        {
            _id: '65a8bda54b191011546c35cf',
            shipId: '65a8bda44b191011546c21b1',
            price: 50000000,
            rentStartDate: '2023-12-09T05:56:51.918Z',
            rentEndDate: '2023-12-14T05:56:51.918Z',
            locationDestination: 'Samarinda',
            locationDeparture: 'Balikpapan',
            source: 'automatic',
            deleteStatus: 'undefined',
            genericDocument: [
                {
                    fileName: 'file1',
                    fileUrl:
                        'https://shiphire-bucket.s3.ap-southeast-1.amazonaws.com/shiphire/shipHistory/65a8bda54b191011546c35cf/file1',
                },
            ],
            needs: '',
            renterCompanyName: '',
        },
    ],
};

const mockParamsEdit = {
    shipHistory: {
        _id: '65a8bda54b191011546c35cf',
        shipId: '65a8bda44b191011546c21b1',
        price: 50000000,
        rentStartDate: '2023-12-09T05:56:51.918Z',
        rentEndDate: '2023-12-14T05:56:51.918Z',
        locationDestination: 'Samarinda',
        locationDeparture: 'Balikpapan',
        source: 'automatic',
        deleteStatus: 'undefined',
        genericDocument: {
            fileName: 'file1',
            fileUrl:
                'https://shiphire-bucket.s3.ap-southeast-1.amazonaws.com/shiphire/shipHistory/65a8bda54b191011546c35cf/file1',
        }[0],
        needs: '',
        renterCompanyName: '',
    },
};

const mockShipData = {
    size: {
        length: 400,
        width: 75,
        height: 20,
    },
    _id: '65b9a9f35f216084af1f9e2f',
    shipOwnerId: {
        company: {
            isRejected: false,
            name: 'Fauzan Company',
            companyType: 'PT',
            address: 'Jl. Raudah',
            bankName: 'BRI',
            bankAccountName: 'Fauzan Corp',
            bankAccountNumber: 123456789,
            documentCompany: [
                {
                    documentName: 'Fauzan Corp Business License',
                    documentUrl:
                        'https://storage.googleapis.com/download/storage/v1/b/shiphire-fdb0e.appspot.com/o/Business%20Document%20test%20company?generation=1695198267402692&alt=media',
                    _id: '65b9a9f35f216084af1f9d2a',
                },
                {
                    documentName: 'Fauzan Corp Deed of Establishment',
                    documentUrl:
                        'https://storage.googleapis.com/download/storage/v1/b/shiphire-fdb0e.appspot.com/o/Deed%20Document%20test%20company?generation=1695198269027762&alt=media',
                    _id: '65b9a9f35f216084af1f9d2b',
                },
            ],
            imageUrl:
                'https://storage.googleapis.com/shiphire-fdb0e.appspot.com/Image%20test%20company?GoogleAccessId=firebase-adminsdk-y4yk5%40shiphire-fdb0e.iam.gserviceaccount.com&Expires=16446988800&Signature=niYNtvjuQFzcv6D%2Bj2h9ULeB6zzFEKk12TYuRAZ4jgYIKAoxFhB0njXXxWiIXuDCb9CCvArAgeLAaXNC47b0XnpfkE7f8KXmy0qEse0Olr168n8TTyvskAm1BSQgtLLHRd7aWJvI75gd3Ia3igKkf43hI1ivqLaoaOSiHz8YSs%2BP1JGjwdgXBBwHQMZlNkokHPaazMyqYIlOdncp4VLjvYfp73JBzOVYfCbam69dr8Jyrn4d5pWT%2Bjqyek24kOJ%2BbvioKc6GyJtynFP7cgpjBASrvtYtkRjma2jwVd3qB80nx%2FopwTO6I3XWunpD9vxrZlt8d%2Fm%2BHhkjrhZd%2FiWL5w%3D%3D',
            isVerified: true,
        },
        _id: '65b9a9f35f216084af1f9d27',
    },
    name: 'Barge Hauler',
    imageUrl:
        'https://storage.googleapis.com/shiphire-fdb0e.appspot.com/Image%20Ship%20Barge%20Hauler?GoogleAccessId=firebase-adminsdk-y4yk5%40shiphire-fdb0e.iam.gserviceaccount.com&Expires=1896710400&Signature=BmuSrc1ujivUWLEr4nH6YW%2BDrF5AAYBC5ILuijc3IZLecUPbajy6p8BF4KbtCJi9n7Ud0UjRvYrW%2FdNXN2S%2Bg2ie1vNu9XOAG0QFRwx03WKcyJ4X4oODpwgiI1j9RLCX2c4vFXAMjvlsqGMoZqLkEEn7pMnLu0lX8RxnZgE5Iv5JH%2FjiRG3LfpcDTmOgf9LECum01nS%2BW%2Bcto7Vs3g2YpanAWCvaq5vvwU3G765WhumhBPRNoQwMAAlqgXdZAEzQ4hqfrF3Fc8V9GmcGQVm2YwL2eNdZmR0djWWGU5gZx6opiCjwJlirmgJfAy4Q%2BB3LRoi7Zhh9xtZdOZm2Tz4j0g%3D%3D',
    desc: 'Barge Hauler is a heavy-duty cargo barge designed to transport goods and materials across rivers, lakes, and coastal areas. Equipped with robust loading and unloading mechanisms, it can handle large and bulky items with ease. Barge Hauler plays a crucial role in facilitating the movement of essential commodities and equipment for various industries.',
    category: {
        _id: '65b9a9f35f216084af1f9d40',
        name: 'Barge',
        __v: 0,
    },
    tags: [],
    shipDocuments: [
        {
            documentName: 'Ship Barge Hauler Document1',
            documentUrl:
                'https://storage.googleapis.com/download/storage/v1/b/shiphire-fdb0e.appspot.com/o/Ship%20Barge%20Hauler%20Owner%20Document%201?generation=1695180758491269&alt=media',
            _id: '65b9a9f35f216084af1f9e30',
        },
        {
            documentName: 'Ship Barge Hauler Document2',
            documentUrl:
                'https://storage.googleapis.com/download/storage/v1/b/shiphire-fdb0e.appspot.com/o/Ship%20Barge%20Hauler%20Owner%20Document%202?generation=1695180762377238&alt=media',
            _id: '65b9a9f35f216084af1f9e31',
        },
    ],
    pricePerMonth: 900000000,
    facilities: [
        {
            type: '65b9a9f35f216084af1f9d4a',
            typeName: 'Cargo Handling Equipment',
            name: 'Cranes',
            _id: '65b9a9f35f216084af1f9e32',
        },
        {
            type: '65b9a9f35f216084af1f9d4a',
            typeName: 'Cargo Handling Equipment',
            name: 'Conveyor Belt',
            _id: '65b9a9f35f216084af1f9e33',
        },
        {
            type: '65b9a9f35f216084af1f9d4c',
            typeName: 'Deck Cranes',
            name: 'Hydraulic Winches',
            _id: '65b9a9f35f216084af1f9e34',
        },
    ],
    specifications: [
        {
            spesificationId: {
                units: 'Units',
            },
            name: 'Vehicle Capacity',
            value: 100,
            _id: '65b9a9f35f216084af1f9e35',
        },
    ],
    rating: 5,
    totalRentalCount: 0,
    shipApproved: true,
    __v: 0,
    rfqDynamicForm: '65b9a9f35f216084af1fa026',
    shipHistory: [
        {
            _id: '65b9a9f35f216084af1fb23d',
            shipId: '65b9a9f35f216084af1f9e2f',
            price: 50000000,
            rentStartDate: '2023-12-22T02:01:22.119Z',
            rentEndDate: '2023-12-27T02:01:22.119Z',
            locationDestination: 'Samarinda',
            locationDeparture: 'Balikpapan',
            source: 'automatic',
            deleteStatus: 'undefined',
            genericDocument: [],
            __v: 0,
        },
        {
            _id: '65b9a9f35f216084af1fb245',
            shipId: '65b9a9f35f216084af1f9e2f',
            price: 50000000,
            rentStartDate: '2024-01-01T02:01:22.119Z',
            rentEndDate: '2024-01-06T02:01:22.119Z',
            locationDestination: 'Samarinda',
            locationDeparture: 'Balikpapan',
            source: 'automatic',
            deleteStatus: 'undefined',
            genericDocument: [],
            __v: 0,
        },
        {
            _id: '65b9a9f35f216084af1fb24d',
            shipId: '65b9a9f35f216084af1f9e2f',
            price: 50000000,
            rentStartDate: '2024-01-11T02:01:22.119Z',
            rentEndDate: '2024-01-16T02:01:22.119Z',
            locationDestination: 'Kutai Timur',
            locationDeparture: 'Kutai Barat',
            source: 'automatic',
            deleteStatus: 'undefined',
            genericDocument: [],
            __v: 0,
        },
    ],
};

const managementTransactionHistoryComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainOwnerStack.Navigator>
                    <MainOwnerStack.Screen
                        name="ManageTransactionHistory"
                        component={ManageTransactionHistory}
                        initialParams={mockParams}
                    />
                    <MainOwnerStack.Screen
                        name="EditShipHistory"
                        component={EditShipHistory}
                        initialParams={mockParamsEdit}
                    />
                </MainOwnerStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

describe('Testing Manage Transaction History Screen', () => {
    describe('Snapshot Testing', () => {
        it('should render Manage Transaction History correctly', async () => {
            const manageTransactionHistory = render(
                managementTransactionHistoryComponent,
            );
            expect(manageTransactionHistory).toMatchSnapshot();
        });
    });

    describe('Unit Testing', () => {
        it('should render item correctly', async () => {
            const manageTransactionHistory = render(
                managementTransactionHistoryComponent,
            );
            const { getByTestId, getByText } = manageTransactionHistory;
            expect(getByTestId('transaction-history-list')).toBeTruthy();
        });

        it('should handle if cancel delete transaction history', async () => {
            const manageTransactionHistory = render(
                managementTransactionHistoryComponent,
            );
            const { getByTestId, getByText } = manageTransactionHistory;
            const deleteButton = getByTestId('delete-transaction-history');
            fireEvent.press(deleteButton);

            const confirmModal = getByTestId('confirm-modal-delete');
            expect(confirmModal).toBeTruthy();

            const cancelButton = getByTestId('cancel-delete-ship-history');
            fireEvent.press(cancelButton);

            expect(getByTestId('transaction-history-list')).toBeTruthy();
        });

        it('should handle error delete transaction history', async () => {
            const manageTransactionHistory = render(
                managementTransactionHistoryComponent,
            );
            const { getByTestId, getByText } = manageTransactionHistory;
            const deleteButton = getByTestId('delete-transaction-history');
            fireEvent.press(deleteButton);

            const confirmModal = getByTestId('confirm-modal-delete');
            expect(confirmModal).toBeTruthy();

            const confirmButton = getByTestId('confirm-delete-ship-history');
            fireEvent.press(confirmButton);

            mockAdapter.onPost('/ship/delete-ship-history').reply(500, {
                message: 'failed',
            });

            await waitFor(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'failed',
                    text: 'ShipOwner.ManageTransactionHistory.errorDelete',
                    visible: true,
                });

                jest.advanceTimersByTime(4000);
            });
        });

        it('should handle success delete transaction history', async () => {
            const manageTransactionHistory = render(
                managementTransactionHistoryComponent,
            );
            const { getByTestId, getByText } = manageTransactionHistory;
            const deleteButton = getByTestId('delete-transaction-history');
            fireEvent.press(deleteButton);

            const confirmModal = getByTestId('confirm-modal-delete');
            expect(confirmModal).toBeTruthy();

            const confirmButton = getByTestId('confirm-delete-ship-history');
            fireEvent.press(confirmButton);

            mockAdapter.onPost('/ship/delete-ship-history').reply(200, {
                message: 'Success',
            });

            await waitFor(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'success',
                    text: 'ShipOwner.ManageTransactionHistory.successDelete',
                    visible: true,
                });

                jest.advanceTimersByTime(4000);

                mockAdapter
                    .onGet(`/ship/get-ship-by-id/${mockParams.shipId}`)
                    .reply(200, {
                        message: 'Success',
                        data: mockShipData,
                    });
            });
        });
    });

    it('should navigate to edit transaction history', async () => {
        const manageTransactionHistory = render(
            managementTransactionHistoryComponent,
        );
        const { getByTestId, getByText } = manageTransactionHistory;
        const editButton = getByTestId('edit-transaction-history');
        fireEvent.press(editButton);

        await waitFor(() => {
            expect(getByTestId('EditHistory')).toBeTruthy();
        });
    });
});
