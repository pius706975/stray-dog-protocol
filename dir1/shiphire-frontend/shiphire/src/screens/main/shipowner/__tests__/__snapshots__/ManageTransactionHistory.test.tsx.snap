// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Manage Transaction History Screen Snapshot Testing should render Manage Transaction History correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <RCTScrollView
              collapsable={false}
              data={
                [
                  "",
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              handlerTag={1}
              handlerType="NativeViewGestureHandler"
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              renderScrollComponent={[Function]}
              scrollEventThrottle={50}
              showsVerticalScrollIndicator={false}
              stickyHeaderIndices={[]}
              style={
                {
                  "height": "100%",
                  "width": "100%",
                }
              }
              viewabilityConfigCallbackPairs={[]}
              waitFor={
                [
                  {
                    "current": null,
                  },
                  {
                    "current": null,
                  },
                ]
              }
            >
              <View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "borderRadius": 8,
                          "height": "100%",
                          "margin": 6,
                        },
                      ]
                    }
                  >
                    <RCTScrollView
                      ItemSeparatorComponent={[Function]}
                      collapsable={false}
                      data={
                        [
                          {
                            "_id": "65a8bda54b191011546c35cf",
                            "deleteStatus": "undefined",
                            "genericDocument": [
                              {
                                "fileName": "file1",
                                "fileUrl": "https://shiphire-bucket.s3.ap-southeast-1.amazonaws.com/shiphire/shipHistory/65a8bda54b191011546c35cf/file1",
                              },
                            ],
                            "locationDeparture": "Balikpapan",
                            "locationDestination": "Samarinda",
                            "needs": "",
                            "price": 50000000,
                            "rentEndDate": "2023-12-14T05:56:51.918Z",
                            "rentStartDate": "2023-12-09T05:56:51.918Z",
                            "renterCompanyName": "",
                            "shipId": "65a8bda44b191011546c21b1",
                            "source": "automatic",
                          },
                        ]
                      }
                      getItem={[Function]}
                      getItemCount={[Function]}
                      handlerTag={2}
                      handlerType="NativeViewGestureHandler"
                      keyExtractor={[Function]}
                      onContentSizeChange={[Function]}
                      onEndReachedThreshold={0.1}
                      onGestureHandlerEvent={[Function]}
                      onGestureHandlerStateChange={[Function]}
                      onLayout={[Function]}
                      onMomentumScrollBegin={[Function]}
                      onMomentumScrollEnd={[Function]}
                      onScroll={[Function]}
                      onScrollBeginDrag={[Function]}
                      onScrollEndDrag={[Function]}
                      removeClippedSubviews={false}
                      renderItem={[Function]}
                      renderScrollComponent={[Function]}
                      scrollEventThrottle={50}
                      stickyHeaderIndices={[]}
                      testID="transaction-history-list"
                      viewabilityConfigCallbackPairs={[]}
                      waitFor={
                        [
                          {
                            "current": null,
                          },
                          {
                            "current": null,
                          },
                        ]
                      }
                    >
                      <View>
                        <View
                          onFocusCapture={[Function]}
                          onLayout={[Function]}
                          style={null}
                        >
                          <View
                            style={
                              [
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                {
                                  "borderBottomWidth": 0.5,
                                  "flexDirection": "row",
                                },
                              ]
                            }
                            testID="transaction-history-item"
                          >
                            <View
                              style={
                                [
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  {
                                    "flex": 1,
                                    "paddingLeft": 14,
                                    "paddingRight": 10,
                                    "paddingVertical": 14,
                                  },
                                ]
                              }
                            >
                              <Text
                                numberOfLines={0}
                                style={
                                  {
                                    "borderBottomWidth": 0,
                                    "borderColor": undefined,
                                    "color": "#4d5963",
                                    "fontFamily": "Kanit-SemiBold",
                                    "fontSize": 14,
                                    "lineHeight": undefined,
                                    "textAlign": undefined,
                                  }
                                }
                              >
                                ShipOwner.ManageTransactionHistory.textRentPeriod
                              </Text>
                              <Text
                                numberOfLines={0}
                                style={
                                  {
                                    "borderBottomWidth": 0,
                                    "borderColor": undefined,
                                    "color": "#4d5963",
                                    "fontFamily": "Kanit-Regular",
                                    "fontSize": 14,
                                    "lineHeight": undefined,
                                    "textAlign": undefined,
                                  }
                                }
                              >
                                Dec 9, 2023
                                 
                                ShipOwner.ManageTransactionHistory.textUntil
                                 
                                Dec 14, 2023
                              </Text>
                              <Text
                                numberOfLines={0}
                                style={
                                  {
                                    "borderBottomWidth": 0,
                                    "borderColor": undefined,
                                    "color": "#4d5963",
                                    "fontFamily": "Kanit-SemiBold",
                                    "fontSize": 14,
                                    "lineHeight": undefined,
                                    "textAlign": undefined,
                                  }
                                }
                              >
                                ShipOwner.ManageTransactionHistory.textLocation
                              </Text>
                              <Text
                                numberOfLines={0}
                                style={
                                  {
                                    "borderBottomWidth": 0,
                                    "borderColor": undefined,
                                    "color": "#4d5963",
                                    "fontFamily": "Kanit-Regular",
                                    "fontSize": 14,
                                    "lineHeight": undefined,
                                    "textAlign": undefined,
                                  }
                                }
                              >
                                Balikpapan
                                 
                                ShipOwner.ManageTransactionHistory.textTo
                                 
                                Samarinda
                              </Text>
                              <View
                                row={true}
                                style={
                                  [
                                    undefined,
                                    undefined,
                                    undefined,
                                    undefined,
                                    {},
                                    {},
                                    {
                                      "flexDirection": "row",
                                    },
                                    {
                                      "flex": 1,
                                      "flexWrap": "wrap",
                                    },
                                  ]
                                }
                              >
                                <View
                                  accessibilityState={
                                    {
                                      "busy": undefined,
                                      "checked": undefined,
                                      "disabled": undefined,
                                      "expanded": undefined,
                                      "selected": undefined,
                                    }
                                  }
                                  accessibilityValue={
                                    {
                                      "max": undefined,
                                      "min": undefined,
                                      "now": undefined,
                                      "text": undefined,
                                    }
                                  }
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "backgroundColor": "#216dab",
                                      "borderRadius": 5,
                                      "flexDirection": "row",
                                      "marginRight": 5,
                                      "marginTop": 5,
                                      "padding": 5,
                                      "paddingHorizontal": 8,
                                    }
                                  }
                                >
                                  <View
                                    marginR-5={true}
                                    style={
                                      [
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        {},
                                        {
                                          "marginRight": 5,
                                        },
                                        {},
                                        undefined,
                                      ]
                                    }
                                  >
                                    <RNSVGSvgView
                                      align="xMidYMid"
                                      bbHeight="20"
                                      bbWidth="16"
                                      fill="none"
                                      focusable={false}
                                      height={20}
                                      meetOrSlice={0}
                                      minX={0}
                                      minY={0}
                                      style={
                                        [
                                          {
                                            "backgroundColor": "transparent",
                                            "borderWidth": 0,
                                          },
                                          {
                                            "flex": 0,
                                            "height": 20,
                                            "width": 16,
                                          },
                                        ]
                                      }
                                      vbHeight={20}
                                      vbWidth={16}
                                      width={16}
                                      xml="<svg width="16" height="20" viewBox="0 0 16 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M10.59 0.59C10.21 0.21 9.7 0 9.17 0H2C0.9 0 0 0.9 0 2V18C0 19.1 0.89 20 1.99 20H14C15.1 20 16 19.1 16 18V6.83C16 6.3 15.79 5.79 15.41 5.42L10.59 0.59ZM11 16H5C4.45 16 4 15.55 4 15C4 14.45 4.45 14 5 14H11C11.55 14 12 14.45 12 15C12 15.55 11.55 16 11 16ZM11 12H5C4.45 12 4 11.55 4 11C4 10.45 4.45 10 5 10H11C11.55 10 12 10.45 12 11C12 11.55 11.55 12 11 12ZM9 6V1.5L14.5 7H10C9.45 7 9 6.55 9 6Z" fill="#fdfdfd"/>
    </svg>
       
    "
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <RNSVGGroup
                                        fill={null}
                                        propList={
                                          [
                                            "fill",
                                          ]
                                        }
                                      >
                                        <RNSVGPath
                                          d="M10.59 0.59C10.21 0.21 9.7 0 9.17 0H2C0.9 0 0 0.9 0 2V18C0 19.1 0.89 20 1.99 20H14C15.1 20 16 19.1 16 18V6.83C16 6.3 15.79 5.79 15.41 5.42L10.59 0.59ZM11 16H5C4.45 16 4 15.55 4 15C4 14.45 4.45 14 5 14H11C11.55 14 12 14.45 12 15C12 15.55 11.55 16 11 16ZM11 12H5C4.45 12 4 11.55 4 11C4 10.45 4.45 10 5 10H11C11.55 10 12 10.45 12 11C12 11.55 11.55 12 11 12ZM9 6V1.5L14.5 7H10C9.45 7 9 6.55 9 6Z"
                                          fill={
                                            {
                                              "payload": 4294835709,
                                              "type": 0,
                                            }
                                          }
                                          propList={
                                            [
                                              "fill",
                                            ]
                                          }
                                        />
                                      </RNSVGGroup>
                                    </RNSVGSvgView>
                                  </View>
                                  <Text
                                    numberOfLines={0}
                                    style={
                                      {
                                        "borderBottomWidth": 0,
                                        "borderColor": undefined,
                                        "color": "#f0f6fb",
                                        "fontFamily": "Kanit-Regular",
                                        "fontSize": 14,
                                        "lineHeight": undefined,
                                        "textAlign": undefined,
                                      }
                                    }
                                  >
                                    file1
                                  </Text>
                                </View>
                              </View>
                            </View>
                            <View
                              style={
                                [
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                  },
                                ]
                              }
                            >
                              <View
                                marginR-5={true}
                                style={
                                  [
                                    undefined,
                                    undefined,
                                    undefined,
                                    undefined,
                                    {},
                                    {
                                      "marginRight": 5,
                                    },
                                    {},
                                    undefined,
                                  ]
                                }
                              >
                                <View
                                  accessibilityState={
                                    {
                                      "busy": undefined,
                                      "checked": undefined,
                                      "disabled": undefined,
                                      "expanded": undefined,
                                      "selected": undefined,
                                    }
                                  }
                                  accessibilityValue={
                                    {
                                      "max": undefined,
                                      "min": undefined,
                                      "now": undefined,
                                      "text": undefined,
                                    }
                                  }
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "alignItems": "center",
                                      "backgroundColor": "#1fcc44",
                                      "borderRadius": 6,
                                      "elevation": 2,
                                      "flexDirection": "row",
                                      "gap": 0,
                                      "justifyContent": "center",
                                      "paddingHorizontal": 10,
                                      "paddingVertical": 6,
                                    }
                                  }
                                  testID="edit-transaction-history"
                                >
                                  <Text
                                    numberOfLines={0}
                                    style={
                                      {
                                        "borderBottomWidth": 0,
                                        "borderColor": undefined,
                                        "color": "#f0f6fb",
                                        "fontFamily": "Kanit-Medium",
                                        "fontSize": 18,
                                        "lineHeight": undefined,
                                        "textAlign": undefined,
                                      }
                                    }
                                  >
                                    ShipOwner.ManageTransactionHistory.buttonEdit
                                  </Text>
                                  <View
                                    style={
                                      [
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        {
                                          "position": "absolute",
                                          "right": 20,
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                              <View
                                marginR-5={true}
                                style={
                                  [
                                    undefined,
                                    undefined,
                                    undefined,
                                    undefined,
                                    {},
                                    {
                                      "marginRight": 5,
                                    },
                                    {},
                                    undefined,
                                  ]
                                }
                              >
                                <View
                                  accessibilityState={
                                    {
                                      "busy": undefined,
                                      "checked": undefined,
                                      "disabled": undefined,
                                      "expanded": undefined,
                                      "selected": undefined,
                                    }
                                  }
                                  accessibilityValue={
                                    {
                                      "max": undefined,
                                      "min": undefined,
                                      "now": undefined,
                                      "text": undefined,
                                    }
                                  }
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "alignItems": "center",
                                      "backgroundColor": "#cc1f12",
                                      "borderRadius": 6,
                                      "elevation": 2,
                                      "flexDirection": "row",
                                      "gap": 0,
                                      "justifyContent": "center",
                                      "paddingHorizontal": 10,
                                      "paddingVertical": 6,
                                    }
                                  }
                                  testID="delete-transaction-history"
                                >
                                  <Text
                                    numberOfLines={0}
                                    style={
                                      {
                                        "borderBottomWidth": 0,
                                        "borderColor": undefined,
                                        "color": "#f0f6fb",
                                        "fontFamily": "Kanit-Medium",
                                        "fontSize": 18,
                                        "lineHeight": undefined,
                                        "textAlign": undefined,
                                      }
                                    }
                                  >
                                    ShipOwner.ManageTransactionHistory.buttonDelete
                                  </Text>
                                  <View
                                    style={
                                      [
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        {
                                          "position": "absolute",
                                          "right": 20,
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </RCTScrollView>
                  </View>
                </View>
              </View>
            </RCTScrollView>
            <Modal
              animationType="slide"
              hardwareAccelerated={false}
              modifiers={
                {
                  "alignments": {},
                  "backgroundColor": undefined,
                  "borderRadius": undefined,
                  "color": "transparent",
                  "flexStyle": undefined,
                  "margins": {},
                  "paddings": {},
                  "positionStyle": undefined,
                  "typography": undefined,
                }
              }
              testID="confirm-modal-delete"
              transparent={true}
              visible={false}
            />
            <Modal
              animationType="slide"
              hardwareAccelerated={false}
              modifiers={
                {
                  "alignments": {},
                  "backgroundColor": undefined,
                  "borderRadius": undefined,
                  "color": "transparent",
                  "flexStyle": undefined,
                  "margins": {},
                  "paddings": {},
                  "positionStyle": undefined,
                  "typography": undefined,
                }
              }
              onRequestClose={[Function]}
              testID="modalPdf"
              transparent={true}
              visible={false}
            />
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="ManageTransactionHistory"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
